src/NosCore.GameObject/Character.cs

```diff
@@ -330,49 +332,49 @@ public async Task LeaveGroupAsync()
+                await _characterDao.TryInsertOrUpdateAsync(character).ConfigureAwait(false);
 
                 var quicklistEntriesToDelete = _quicklistEntriesDao
                     .Where(i => i.CharacterId == CharacterId).ToList()
                     .Where(i => QuicklistEntries.All(o => o.Id != i.Id)).ToList();
-                _quicklistEntriesDao.Delete(quicklistEntriesToDelete.Select(s => s.Id).ToArray());
-                _quicklistEntriesDao.InsertOrUpdate(QuicklistEntries);
+                await _quicklistEntriesDao.TryDeleteAsync(quicklistEntriesToDelete.Select(s => s.Id).ToArray()).ConfigureAwait(false);
+                await _quicklistEntriesDao.TryInsertOrUpdateAsync(QuicklistEntries).ConfigureAwait(false);
 
                 // load and concat inventory with equipment
                 var itemsToDelete = _inventoryItemInstanceDao
                     .Where(i => i.CharacterId == CharacterId).ToList()
                     .Where(i => InventoryService.Values.All(o => o.Id != i.Id)).ToList();
 
-                _inventoryItemInstanceDao.Delete(itemsToDelete);
-                _itemInstanceDao.Delete(itemsToDelete.Select(s => s.ItemInstanceId).ToArray());
+                await _inventoryItemInstanceDao.TryDeleteAsync(itemsToDelete.Select(s => s.Id).ToArray()).ConfigureAwait(false);
+                await _itemInstanceDao.TryDeleteAsync(itemsToDelete.Select(s => s.ItemInstanceId).ToArray()).ConfigureAwait(false);
 
-                _itemInstanceDao.InsertOrUpdate(InventoryService.Values.Select(s => s.ItemInstance!).ToArray());
-                _inventoryItemInstanceDao.InsertOrUpdate(InventoryService.Values.ToArray());
+                await _itemInstanceDao.TryInsertOrUpdateAsync(InventoryService.Values.Select(s => s.ItemInstance!).ToArray()).ConfigureAwait(false);
+                await _inventoryItemInstanceDao.TryInsertOrUpdateAsync(InventoryService.Values.ToArray()).ConfigureAwait(false);
 
                 var staticBonusToDelete = _staticBonusDao
                     .Where(i => i.CharacterId == CharacterId).ToList()
                     .Where(i => StaticBonusList.All(o => o.StaticBonusId != i.StaticBonusId)).ToList();
-                _staticBonusDao.Delete(staticBonusToDelete);
-                _staticBonusDao.InsertOrUpdate(StaticBonusList);
+                await _staticBonusDao.TryDeleteAsync(staticBonusToDelete.Select(s => s.StaticBonusId)).ConfigureAwait(false);
+                await _staticBonusDao.TryInsertOrUpdateAsync(StaticBonusList).ConfigureAwait(false);
 
-                _titleDao.InsertOrUpdate(Titles);
+                await _titleDao.TryInsertOrUpdateAsync(Titles).ConfigureAwait(false);
 
                 var minilandDto = (MinilandDto)_minilandProvider.GetMiniland(CharacterId);
-                _minilandDao.InsertOrUpdate(ref minilandDto);
+                minilandDto = await _minilandDao.TryInsertOrUpdateAsync(minilandDto).ConfigureAwait(false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'minilandDto'.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5326532)